Wangscape/tilegen/TilesetGenerator.cpp

```diff
@@ -30,21 +40,31 @@ void TilesetGenerator::generate(std::function<void(const sf::Texture&, std::stri
     {
         CornersGenerator corners_generator(clique, static_cast<size_t>(CORNERS));
         std::pair<size_t, size_t> tileset_resolution = corners_generator.size2D(options.tileFormat.resolution);
-        size_t res_x = tileset_resolution.first;
-        size_t res_y = tileset_resolution.second;
-        std::unique_ptr<sf::RenderTexture> output{getBlankImage(res_x, res_y)};
+        mResX = tileset_resolution.first;
+        mResY = tileset_resolution.second;
+        std::unique_ptr<sf::RenderTexture> output{getBlankImage(mResX, mResY)};
+        mDebugTilesets.clear();
 
-        auto filename = getOutputImageFilename(clique);
+        auto filename_base = getOutputImageFilename(clique);
+        auto filename = filename_base + options.tileFormat.fileType;
 
         // MetaOutput.addTileset, addTile should use this version of filename;
         // relative to output dir, not options dir!
-        metaOutput.addTileset(clique, filename, res_x, res_y);
+        metaOutput.addTileset(clique, filename, mResX, mResY);
         generateClique(clique, *output, filename);
         output->display();
 
         p.append(filename);
         callback(output->getTexture(), p.string());
         p.remove_filename();
+        for (auto& it : mDebugTilesets)
```

Comment:
It compiles and seems to work for me. The thing is, if I reason correctly, with `const auto&` you'll have a const reference to the `std::pair<DebugTilesetID, std::unique_ptr<sf::RenderTexture>>` and while you **cannot** call any non const method that will change value of this **unique ptr**, methods of underlying object don't change a value of it, so it's not a problem. Have you tried it on you compiler?

---

Example
```cpp
#include <iostream>
#include <memory>

struct A
{
    void fx() {++x_;}
    int x_ = 0;
};

struct B
{
    B() : a{std::make_unique<A>()}
    {}
    std::unique_ptr<A> a;
    int y_ = 0;
};

int main()
{
    const B b1;
    const B& b2 = b1;
    std::cout << b2.y_ << std::endl;
    b2.a->fx();
    std::cout << b2.a->x_ << std::endl;
}
```

On `gcc 7.1.1` and `clang 4.0.1` this snippet produces:

```shell
0
1
```